float32 f_pwm_rate
float32 f_pwm_offset
float32 m_f_rate
float32 pwm_rate #percentage
